python/Scripts/creatematerial.py

```diff
@@ -0,0 +1,235 @@
+# -*- coding: utf-8 -*-
+__doc__ = 'Create materialX file conatin standard surface uber shader with input texture for given directory'
+
+import os
+import re
+import json
+import argparse
+
+import logging
+from typing import List, Dict
+
+import MaterialX
+import udimutils
+
+# Configure the logger
+logging.basicConfig(
+    level=logging.INFO,
+    format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+)
+
+# Create a logger instance
+logger = logging.getLogger('creatematerial')
+
+
+def get_cfg() -> Dict:
+    """
+    Get the patterns and settings from cfg file
+    return: dict of resultant settings
+    """
+    json_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'textures_patterns.json')
+    # TODO: Add option here to take the user overrides customization file
+    with open(json_path, 'r') as f:
+        data = json.load(f)
+    return data
+
+
+def texture_type_from_name(texture_name: str) -> List[str]:
+    """
+    Get the texture matched pattern and return the materialx plug name and the plug type
+    @param texture_name: The texture filename
+    return: list(material_input_name, material_input_type)
+    """
+    settings = get_cfg()
+    texture_patterns = settings.get('patterns')
+
+    # reversed to take the color before weight e.g. specular_color before specular
+    for plug_name in reversed(list(texture_patterns.keys())):
+        regx_list = texture_patterns[plug_name]['values']
+        if not isinstance(regx_list, list):
+            continue
+
+        for regex in regx_list:
+            if re.search(regex, texture_name):
+                return [plug_name, texture_patterns[plug_name]['type']]
+
+
+def list_textures(texture_dir: MaterialX.FilePath) -> List[udimutils.UDIMFile]:
```

Comment:
Perhaps something for later consideration: if a series of images represents an animated texture that token parsing could be added ? e.g. `baseColor.1.exr, baseColor.2.exr`.

Reply:
Good point. Do the animated textures take a different token from `<UDIM>` and what if we have UDIM animated textures?